app/models/ex_link.rb

```diff
@@ -0,0 +1,61 @@
+class ExLink < ActiveRecord::Base
+  validates :label, :url, presence: true, uniqueness: true
+  has_many :ex_link_tags
+  has_many :tags, through: :ex_link_tags
+
+  attr_accessor :tagstring
+
+  before_save :add_tags
+
+  has_attached_file :image,
```

Comment:
I have tried to use CarrierWave instead, and I find it rather nice to use.
It is implemented as this:
`mount_uploader :image, AttachedImageUploader` where `image` is a string field, easier than the paperclip attachment.